src/app/keystore/page.tsx

```
@@ -0,0 +1,1109 @@
+
+    // Get proof using our proxy API - this will now use the direct RPC call
+    const proofData = await callAxiomKeystoreRPC('axiom_getProof', {
+      address: keystoreAddress,
+    });
+
+    // Check if we have a valid response
+    if (proofData && proofData.state && proofData.proof) {
+      // Extract data from the response
+      const { state, proof } = proofData;
+
+      // Construct isLeft bitmap - a bit packed form of the isLeft flags
+      const isLeft = proof.siblings.reduce((acc: bigint, sibling: KeystoreSibling, index: number) => {
+        return acc | (BigInt(sibling.isLeft ? 1 : 0) << BigInt(index));
+      }, BigInt(0));
+
+      // Extract the siblings hashes for the proof array
+      const proofArray = proof.siblings.map((sibling: KeystoreSibling) => sibling.hash);
+
+      // Get the leaf data
+      const leaf = proof.leaf;
+
+      // For exclusion proofs, construct exclusionExtraData
+      // exclusionExtraData: abi.encodePacked(
+      //    bytes1(keyPrefix),
+      //    bytes32(key),
+      //    salt,
+      //    keccak256(abi.encode(bytes32(value)))
+      // )
+
+      // Calculate valueHash using abi.encode wrapped in keccak256
+      const valueHash = keccak256(encodeAbiParameters([{ type: 'bytes32' }], [leaf.value]));
+
+      // Construct exclusionExtraData
+      const exclusionExtraData = concat([leaf.keyPrefix, leaf.key, userSalt, valueHash]) as `0x${string}`;
+
+      return {
+        isExclusion: proof.isExclusionProof,
+        exclusionExtraData,
+        vkeyHash: VKEY_HASH,
+        keyData: '0x' as `0x${string}`,
+        proof: proofArray,
+        isLeft,
+        nextDummyByte: leaf.nextKeyPrefix,
+        nextImtKey: leaf.nextKey,
+      };
+    }
+
+    // If something went wrong with parsing or the response is invalid, log and fall back to mock data
+    console.warn('Could not parse Axiom response correctly, falling back to mock data');
+
+    // Fallback to mock data if needed
```

this wont work though right? surely we should return an error here?